R/openmp.R

```diff
@@ -0,0 +1,22 @@
+## We should have the system report back if it supports openmp at all,
+## and pass that in here too, because that deserves a warning.
+##
+## We should also pass in the number of cores available and warn if
+## the user goes over this.
+##
```

Comment:
Yes - thinking about cluster usage too - whether/how best to detect allotted cores, not just machine cores.

Reply:
we have code for this in dust, but I was planning on doing it as another PR for now because it's actually quite separate - it's also going to interact strongly with the way we do things from mcstate in parallel